Meta/Lagom/Tools/CodeGenerators/LibWeb/GenerateAriaRoles.cpp

```diff
@@ -0,0 +1,397 @@
+        bool first = true;
+        TRY(super_classes.try_for_each([&](JsonValue const& value) -> ErrorOr<void> {
+            VERIFY(value.is_string());
+
+            class_definition_generator.append(first ? " "sv : ", "sv);
+            class_definition_generator.append(TRY(String::formatted("public {}", value.as_string())));
+            first = false;
+            return {};
+        }));
+
+        class_definition_generator.append(R"~~~(
+{
+public:
+    @name@(AriaData const&);
+
+    virtual HashTable<StateAndProperties> supported_states() const override;
+    virtual HashTable<StateAndProperties> supported_properties() const override;
+
+    virtual HashTable<StateAndProperties> required_states() const override;
+    virtual HashTable<StateAndProperties> required_properties() const override;
+
+    virtual HashTable<StateAndProperties> prohibited_properties() const override;
+    virtual HashTable<StateAndProperties> prohibited_states() const override;
+
+    virtual HashTable<Role> required_context_roles() const override;
+    virtual HashTable<Role> required_owned_elements() const override;
+    virtual bool accessible_name_required() const override;
+    virtual bool children_are_presentational() const override;
+    virtual DefaultValueType default_value_for_property_or_state(StateAndProperties) const override;
+protected:
+    @name@();
+)~~~");
+
+        auto name_from_source = value.as_object().get("nameFromSource"sv).value();
+        if (!name_from_source.is_null())
+            class_definition_generator.append(R"~~~(
+public:
+    virtual NameFromSource name_from_source() const override;
+)~~~");
+        class_definition_generator.appendln("};");
+        return {};
+    }));
+
+    generator.appendln("}");
+
+    TRY(file.write_until_depleted((generator.as_string_view().bytes())));
+    return {};
+}
+
+ErrorOr<String> generate_hash_table_population(JsonArray const& values, StringView hash_table_name, StringView enum_class)
+{
+    StringBuilder builder;
+    TRY(values.try_for_each([&](auto& value) -> ErrorOr<void> {
+        VERIFY(value.is_string());
+        TRY(builder.try_appendff("        {}.set({}::{});\n", hash_table_name, enum_class, value.as_string()));
+        return {};
+    }));
+
+    return builder.to_string();
+}
+
+ErrorOr<void> generate_hash_table_member(SourceGenerator& generator, StringView member_name, StringView hash_table_name, StringView enum_class, JsonArray const& values)
+{
+    auto member_generator = TRY(generator.fork());
+    member_generator.set("member_name"sv, member_name);
+    member_generator.set("hash_table_name"sv, hash_table_name);
+    member_generator.set("enum_class"sv, enum_class);
+    TRY(member_generator.set("hash_table_size"sv, TRY(String::number(values.size()))));
+
+    if (values.size() == 0) {
+        member_generator.append(R"~~~(
+HashTable<@enum_class@> @name@::@member_name@() const
```

Comment:
This still returns a copy of the HashTable. You can return a reference to it instead to avoid the copy each time it's called:

```suggestion
HashTable<@enum_class@> const& @name@::@member_name@() const
```

(And same change below for the implementation.)